lsl_autobids/processing_new_files.py

```diff
@@ -1,154 +1,123 @@
-            processed_files.append(file_name_without_ext + ext)
-    print("The processed files are: \n")
-    print(processed_files)
+            file_name_no_ext, ext = os.path.splitext(file_name)
+            if file_name_no_ext.endswith('_old'):
+                user_choice = get_user_input("WARNING: The file appears to be a duplicate ('_old'). Overwrite?")
+                if user_choice == 'n':
+                    logger.warning("User declined to overwrite file. Exiting.")
+                    raise RuntimeError("File overwrite declined by user.")
+                logger.info("Overwrite confirmed.")
+                continue 
+            processed_files.append(file_name_no_ext + ext)
+    
+    logger.info(f"Processed files: {processed_files}")
 
-    project_path = os.path.join(project_root,project_name)
     # User prompt asking if we want to proceed to convert and upload
-    ask_convert_message = "Do you want to proceed for BIDS Conversion? (y/n): "
-    while True:
-        user_response = input(ask_convert_message)
-        
-        if user_response.lower() == 'n':
-            warnings.warn("Operation aborted. Files will not be converted.", UserWarning)
-            file = os.path.join(project_path,"last_run_log.txt")
-            with open(file, 'w') as file:
-                file.truncate(0)
-            sys.exit()
-        elif user_response.lower() == 'y':
-            print('Operation resumed. Files will be converted to BIDS.')
-            break
-        else:
-            print('Invalid response. Please enter "y" for yes or "n" for no.')
-   
-    bids_process_and_upload(processed_files, project_name)
+    if cli_args.yes:
+        # Automatically proceed without asking
+        logger.info("Automatically proceeding with BIDS conversion.")
+        logger.info("Starting BIDS conversion.")
+        bids_process_and_upload(processed_files)
+    else:
+        user_choice = get_user_input("Do you want to proceed with BIDS conversion?")
+        if user_choice == 'n':
+            logger.warning("User aborted BIDS conversion.")
+            _clear_last_run_log(project_name)
+            raise RuntimeError("BIDS conversion aborted by user.")
 
-    
+        logger.info("Starting BIDS conversion.")
+        bids_process_and_upload(processed_files)
 
-def check_for_new_files(function_path):
 
+def _clear_last_run_log() -> None:
+    """Clears the last run log file for the given project.
     """
-    Checks for new files in a deep folder structure since the last run.
-    
-    Parameters:
-    -----------
-    function_path : str
-        The path to the directory to check for new files
-
-    Returns:    
-    --------
-    list
-        A list of new files found since the last run
+    project_name = cli_args.project_name
+    log_path = os.path.join(project_root, project_name, "last_run_log.txt")
+    try:
+        with open(log_path, 'w') as f:
+            f.truncate(0)
+        logger.info("Cleared last run log.")
```

Comment:
If we clear it, and the run fails, wouldn't we repeat the process for _all_ files the next time after?

Reply:
Ah I think this is the actual end of the program, right?